prometheus/src/main/java/org/opensearch/sql/prometheus/client/PrometheusClientImpl.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.prometheus.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Objects;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONObject;
+
+public class PrometheusClientImpl implements PrometheusClient {
+
+  private static final Logger logger = LogManager.getLogger(PrometheusClientImpl.class);
+
+  private final OkHttpClient okHttpClient;
+
+  private final URI uri;
+
+  public PrometheusClientImpl(OkHttpClient okHttpClient, URI uri) {
+    this.okHttpClient = okHttpClient;
+    this.uri = uri;
+  }
+
+
+  @Override
+  public JSONObject queryRange(String query, long start, long end, String step) throws IOException {
+    HttpUrl httpUrl = new HttpUrl.Builder()
+        .scheme(uri.getScheme())
+        .host(uri.getHost())
+        .port(uri.getPort())
+        .addPathSegment("api")
+        .addPathSegment("v1")
+        .addPathSegment("query_range")
+        .addQueryParameter("query", query)
+        .addQueryParameter("start", Long.toString(start))
+        .addQueryParameter("end", Long.toString(end))
+        .addQueryParameter("step", step)
+        .build();
+    logger.info("queryUrl: " + httpUrl);
+    Request request = new Request.Builder()
+        .url(httpUrl)
+        .build();
+    Response response = this.okHttpClient.newCall(request).execute();
```

Comment:
I have not used prometheus much and not a big difference, but any reason to construct the http request manually instead of using a client library like https://github.com/mckang/promql_java_client or https://github.com/anhdat/prometheus4j?

Reply:
There are no official client libraries. Moreover, prometheus provides a simple rest interface with queryString and time range. Since the query interface is not complicated, don't want to introduce any third party library.